scripts/lint_translations.clj

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env bb
+
+(ns lint-translations
+  (:require [babashka.pods :as pods]))
+
+(pods/load-pod 'clj-kondo/clj-kondo "2023.09.07")
+(require '[pod.borkdude.clj-kondo :as kondo])
+(require '[cheshire.core :as json])
+(require '[clojure.set :as set])
+
+(def src-paths ["src"])
+
+(def translation-file "translations/en.json")
+
+;; set the following to true when solving https://github.com/status-im/status-mobile/issues/17811
+(def flag-show-non-namespaced-translation-keys false)
+(def flag-show-non-namespaced-translation-keys-occurrences false) ;; this makes the output super verbose!
+
+;; set the following to true when solving https://github.com/status-im/status-mobile/issues/17813
+;; and keep it permanently on after #17811 and #17813 have both been solved
+(def flag-show-unused-translation-keys false)
+
+(def flag-show-missing-translation-keys true)
+
+(defn- safe-name
+  [x]
+  (when x (name x)))
+
+(defn- ->keyword
+  [analysis-keyword]
+  (keyword (safe-name (:ns analysis-keyword)) (:name analysis-keyword)))
+
+(defn- report-issues
+  [incorrect-usages]
+  (doseq [incorrect-usage incorrect-usages]
+    (->> incorrect-usage
+         ((juxt :filename :row :reason ->keyword))
+         (apply format "%s:%s %s %s")
+         println)))
+
+(defn- extract-translation-keys
+  [file]
+  (-> file slurp json/parse-string keys))
+
+
+(def ^:private probably-unused-warning
+  (format "Probably unused translation key in %s:" translation-file))
+
+(defn -main
+  [& _args]
+  (let [result                           (kondo/run!
+                                          {:lint   src-paths
+                                           :config {:output {:analysis {:keywords true}}}})
+        all-keywords                     (get-in result [:analysis :keywords])
+        used-translations                (filter (comp (partial = 't) :ns) all-keywords)
+        file-translation-keys            (apply sorted-set (extract-translation-keys translation-file))
+        missing-translations             (remove (comp file-translation-keys :name) used-translations)
+        used-translation-keys            (set (map :name used-translations))
+        possibly-unused-translation-keys (set/difference file-translation-keys used-translation-keys)
+        non-namespaced-translations      (filter
+                                          (fn [kw]
+                                            (and (not (:ns kw))
+                                                 (possibly-unused-translation-keys (:name kw))))
+                                          all-keywords)
+        unused-translation-keys          (set/difference possibly-unused-translation-keys
+                                                         (set (map :name non-namespaced-translations)))]
+
+    ;; delete the following once #17811 and #17813 have both been solved
```

Comment:
If you can, please change the actionable comments to use `TODO(akatov): ...`, or like we prefer, move the TODO comments to their respective issues. 

> _TODO_ comments are used extensively in the codebase, but prefer to use them
only when strictly necessary and when an issue is not enough to track the work
left to be done.
>
> -- https://github.com/akatov/status-mobile/blob/develop/doc/new-guidelines.md#using-todos-comments

Reply:
Thanks 👍🏼  We try to make TODOs and issues usually mutually exclusive, maybe that's not clear in the guidelines. If the developer created the issue and properly added the context as you did, we prefer to not have TODO comments. It's not wrong, but we strongly gravitate towards only issues.